mantle-build/src/rpc.rs

```diff
@@ -266,31 +255,29 @@ fn convert_ty(tcx: TyCtxt, ty: &hir::Ty) -> Result<Type, UnsupportedTypeError> {
         ),
         TyKind::Path(hir::QPath::Resolved(_, path)) => {
             use hir::def::{DefKind, Res};
+            let type_args = crate::utils::get_type_args(&path);
             match path.res {
-                Res::Def(kind, id) => match kind {
+                Res::Def(kind, did) => match kind {
                     DefKind::Struct
                     | DefKind::Union
                     | DefKind::Enum
                     | DefKind::Variant
-                    | DefKind::TyAlias
                     | DefKind::Const => {
-                        let type_args = crate::utils::get_type_args(&path);
-                        let is_vec_u8 = |vec_ty: &hir::Ty| match vec_ty.node {
-                            hir::TyKind::Path(hir::QPath::Resolved(_, ref path))
-                                if path.to_string() == "u8" =>
-                            {
-                                true
-                            }
-                            _ => false,
-                        };
-                        convert_def!(
-                            tcx,
-                            id,
-                            ty.hir_id.owner_local_def_id().to_def_id(),
-                            |tcx, _, ty| convert_ty(tcx, ty),
-                            |i| { type_args[i] },
-                            is_vec_u8
-                        )?
+                        convert_def!(tcx, did, ty.hir_id.owner_local_def_id().to_def_id(), |i| {
+                            convert_ty(tcx, type_args[i])
+                        })?
+                    }
+                    DefKind::TyAlias => {
+                        convert_sty_with_arg_at(tcx, did, tcx.type_of(did), |substs, i| {
+                            // Use the HIR type parameter, if it exists.
+                            // Otherwise, the TyS has extra generics that were defined
+                            // in the alias as concrete types (and will be present in the
+                            // TyS substs).
```

Comment:
example:
```rust
type Result<T> = std::result::Result<T, String>;
```
the type of `T` must come from the HIR since the `AdtDef` only knows about _placeholder_ `T`. The `AdtDef` does have the `String`, though, since it's part of the typedef.